fvm/state/state.go

```diff
@@ -128,13 +126,12 @@ func (s *State) Get(owner, key string, enforceLimit bool) (flow.RegisterValue, e
 		return nil, fmt.Errorf("failed to read key %s on account %s: %w", PrintableKey(key), hex.EncodeToString([]byte(owner)), getError)
 	}
 
-	// if not part of recent updates count them as read
-	if _, ok := s.updateSize[mapKey{owner, key}]; !ok {
-		s.ReadCounter++
-		s.TotalBytesRead += uint64(len(owner) + len(key) + len(value))
-	}
-
 	if enforceLimit {
+		// if not part of recent updates count them as read
+		if _, ok := s.updateSize[mapKey{owner, key}]; !ok {
+			s.TotalBytesRead += uint64(len(owner) + len(key) + len(value))
+		}
+
```

Comment:
I'm not sure if thats what we want to do, ideally we want to be able to measure interactions but not enforce them.

Reply:
if that's the case, shouldn't we also do the same for other metered metrics?  this is consistent with computation / memory metering.

Reply:
we shouldn't confuse metering with observability metrics.  perhaps the right thing to do is to add a separate prometheus metric?

Reply:
> computation / memory metering (i.e., we don't count when limit is not enforced). 

I mean technically we are counting when weight is zero: https://github.com/onflow/flow-go/blob/67ed1ebcc41aa6b7cf39677d58500bbf7251c6be/fvm/meter/weighted_meter.go#L354

Reply:
Like Ramtin said, we still want to measure we just don't want there to be a limit with interactions.

Its actually computation and memory that should behave the same way as the interaction. But the `enforceMemoryLimits` and `enforceComputationLimits` were used to get around some bugs, so we need to write some tests for that before we change the behavior.

Reply:
All I'm advocating for is consistency between how we measure and limit thing.  How that's achieved doesn't matter much to me.

With that say, changing interaction limit to behave like memory/computation limits will cause us less headache in the long run, since otherwise, we will need to readjust the limit threshold every time we make major change to the non-enforced portion of the execution.

As I've pointed out above, we shouldn't conflict observability metrics with limit enforcement.  e.g., we can just maintain two meters; a unlimited meter for measuring everything and a limited mete for limit enforcement.